### PR TITLE
Clean `generic.py` and `core.py`

### PR DESCRIPTION
**Pull request by samwaseda**
This PR is mostly about changing `if ID` to `if ID is not None`. So reviewing this PR should be relatively straightforward. There is just one function that I added: `remove_and_reset_id`, which should remove redundant lines in `kill`, `run` and `run_job_with_status_finished`

Edit: following [my comment](https://github.com/pyiron/pyiron_base/issues/1033), I removed the lines in `run_job_with_status_finished`

### CODE REVIEW

File: pyiron_base/jobs/job/generic.py [hunk 1]
```diff
@@ -990,10 +988,16 @@ def job_file_name(self, file_name, cwd=None):
         Returns:
             str: absolute path to the file in the current working directory
         """
-        if not cwd:
+        if cwd is None:
             cwd = self.project_hdf5.working_directory
         return posixpath.join(cwd, file_name)
 
+    def _set_hdf(self, hdf=None, group_name=None)
```
Comment (jan-janssen) [inline pyiron_base/jobs/job/generic.py]:
```suggestion
    def _set_hdf(self, hdf=None, group_name=None):
```

**Comment by samwaseda** [comment]
https://github.com/pyiron/pyiron_base/blob/104460b5a9b5c55a9842275624b8d0c7d46775da/pyiron_base/jobs/job/core.py#L829-L839

It sounds like it should raise `NotImplementedError`, but in reality it is defined here. Can I move the current lines to `GenericJob` and make the one in `JobCore` raise `NotImplementedError`? @jan-janssen

**Comment by jan-janssen** [comment]
> It sounds like it should raise `NotImplementedError`, but in reality it is defined here. Can I move the current lines to `GenericJob` and make the one in `JobCore` raise `NotImplementedError`? @jan-janssen

As the `job_id` is defined on the `JobCore` level and `reset_job_id()` is used in the `copy()` function and the `_internal_copy_to()` function, I guess it is not a good idea to move it right now. Still if you want to move it, I would suggest making a separate pull request, so we can analyze the dependencies in more detail.

**Comment by jan-janssen** [comment]
@samwaseda To me this again is a pull request with too many different changes. It really simplifies the reviewing process if you make separate pull requests for changes that belong together. In this case the introduction of the `remove_and_reset_id(self, _protect_childs=True)` function is a separate topic which should be a separate pull request to make it easier for the reviewers to see which changes are related to this change.

**Comment by samwaseda** [comment]
> @samwaseda To me this again is a pull request with too many different changes. It really simplifies the reviewing process if you make separate pull requests for changes that belong together. In this case the introduction of the `remove_and_reset_id(self, _protect_childs=True)` function is a separate topic which should be a separate pull request to make it easier for the reviewers to see which changes are related to this change.

Last time you specifically said 100 lines, and I was sticking to it in this PR + I did not create a new functionality; I only removed redundant lines and `if None` statements. I don't really understand your criterion of "too many changes", so I'm gonna close this one

**Comment by jan-janssen** [comment]
> Last time you specifically said 100 lines, and I was sticking to it in this PR + I did not create a new functionality; I only removed redundant lines and `if None` statements. I don't really understand your criterion of "too many changes", so I'm gonna close this one

I cannot really follow this reaction. Just by separating this pull request into to https://github.com/pyiron/pyiron_base/pull/1035 and https://github.com/pyiron/pyiron_base/pull/1036 it becomes much more obvious which changes belong together.

**Pull request closed without merging**